controller/src/endpoints/post.py

```diff
@@ -1,37 +1,239 @@
+            log.error(f"Error in GET /posts: {e}")
+            return {"message": "Failed to fetch posts"}, 500
+
+    @jwt_required()
+    @api.doc(
+        description="Create a new post for the authenticated user.",
+        consumes=["multipart/form-data"],
+        params={
+            "description": {"description": "Text description of the post.", "example": "My first post!", "required": True},
+            "images": {"description": "List of image files to upload.", "type": "file", "required": True, "example": "image.jpg"}
+        }
+    )
+    @api.response(201, "Post created successfully")
+    @api.response(400, "Invalid input")
+    @api.response(404, "User not found")
+    @api.response(500, "Internal server error")
+    @api.marshal_with(post_model, code=201)
+    def post(self):
+        """
+        Create a new post for the authenticated user.
+        """
+        log.info("POST /posts endpoint hit.")
+        user_id = get_jwt_identity()
+        log.info(f"Authenticated user ID: {user_id}")
+        queries = db()
+
+        try:
+
+            user = queries.get_user_by_id(user_id)
+            if not user:
+                log.error(f"User not found for ID: {user_id}")
+                api.abort(404, "User not found")
+
+
+            description = request.form.get('description', '').strip()
+            if not description:
+                log.error("Description cannot be empty.")
+                api.abort(400, "Description cannot be empty")
+
 
+            files = request.files.getlist('images')
+            if not files:
+                log.error("No image files provided.")
+                api.abort(400, "No image files provided")
+
+            # Upload files to Imgur
+            imgur_urls = []
+            for file in files:
+                try:
+                    headers = {"Authorization": f"Client-ID {os.environ.get('IMGUR_CLIENT_ID')}"}
+                    log.info(f"Uploading file to Imgur: {file.filename}")
+                    response = send_request('POST', Url.IMGUR, files={'image': file}, headers=headers)
+                except Exception as e:
+                    log.error(f"Error uploading to Imgur: {e}")
+                    api.abort(500, "Failed to upload image to Imgur")
+
+                if response.status_code != 200:
+                    log.error(f"Imgur upload failed: {response.text}")
+                    api.abort(500, "Imgur upload failed")
+
+                imgur_data = response.json()
+                imgur_urls.append(imgur_data['data']['link'])
+                log.info(f"Uploaded to Imgur: {imgur_urls[-1]}")
+            
+            location = user.get("location", "").strip()
+
+            # Create new post
+            new_post = {
+                "_user_id": user_id,
+                "description": description,
+                "images_urls": imgur_urls,
+                "timestamp": datetime.utcnow(),
+                "reactions_count": {"likes": 0},
```

Comment:
Jak ostatnio robiłem endpointy do reakcji to zdałem sobie sprawę, że najlepiej będzie przechowywać reakcje jako np osobna kolekcja, bo stosując reactions_count chyba nie możemy śledzić, czy dany użytkownik już dał reakcję pod postem, więc bedzie mógł dawać w nieskonczoność.
Ja bym proponował, aby przechowywać w kolekcji posts reactions_id które będą wskazywać na _id w kolekcji reactions. Tam bedziemy mogli wyciągać np kto dał reakcję i czy dany uzytkownik już dał reakcję i jaką.
Ponadto dodałbym jeszcze pole comments_id i tam przechowywać wszystkie _id komentarzy które są pod tym postem.